pkg/kube/ready_test.go

```diff
@@ -521,6 +717,144 @@ func Test_ReadyChecker_serviceReady(t *testing.T) {
 	}
 }
 
+func Test_ReadyChecker_crdBetaReady(t *testing.T) {
```

Comment:
I see there are different test functions for different types against the same `ReadyChecker`. The different functions are testing different internal functions. Should these all be run through `IsReady` (the public function that calls these internal ones)? Thoughts?

Reply:
You are correct, I don't like how this is written currently but I wanted to cover with test first then start refactoring. As I don't want to break anything. IsReady is huge and needs to be paired down.